mm2src/rpc_task/src/rpc_common.rs

```diff
@@ -47,6 +51,43 @@ impl HttpStatusCode for RpcTaskUserActionError {
     }
 }
 
+/// In most cases, the RPC task status request may fail with the following errors.
+/// Please do not add new error variants unless they are used in most cases.
+#[derive(Display, Serialize, SerializeErrorType)]
+#[serde(tag = "error_type", content = "error_data")]
+pub enum CancelRpcTaskError {
+    #[display(fmt = "No such task '{}'", _0)]
+    NoSuchTask(TaskId),
+    #[display(fmt = "Task is finished already")]
+    TaskFinished(TaskId),
+    #[display(fmt = "Internal error: {}", _0)]
+    Internal(String),
+}
+
+impl From<RpcTaskError> for CancelRpcTaskError {
+    fn from(e: RpcTaskError) -> Self {
+        match e {
+            RpcTaskError::NoSuchTask(task_id) => CancelRpcTaskError::NoSuchTask(task_id),
+            RpcTaskError::UnexpectedTaskStatus {
+                task_id,
+                actual,
+                expected,
+                ..
+            } if actual.is_finished() && !expected.is_finished() => CancelRpcTaskError::TaskFinished(task_id),
+            rpc_err => CancelRpcTaskError::Internal(rpc_err.to_string()),
+        }
+    }
+}
+
+impl HttpStatusCode for CancelRpcTaskError {
+    fn status_code(&self) -> StatusCode {
+        match self {
+            CancelRpcTaskError::NoSuchTask(_) | CancelRpcTaskError::TaskFinished(_) => StatusCode::BAD_REQUEST,
```

Comment:
Maybe it's better for `CancelRpcTaskError::TaskFinished` status code to be `409 Conflict`. I am not really sure, What is your opinion @sergeyboyko0791?

Reply:
Good idea, done